### PR TITLE
Fix stokes residual postprocessor for purely expensive solves

### PR DESCRIPTION
**Pull request by gassmoeller**
`SolverControl::get_history_data` throws an exception if asked for non-existent history data (like if we do not make any cheap Stokes solver steps). Make sure there is history data for the postprocessor to use before calling that function. The new test fails with current master, but succeeds with the patch. Still need to update the test results.

### CODE REVIEW

File: source/postprocess/stokes_residual.cc [hunk 1]
```diff
@@ -106,7 +107,8 @@ namespace aspect
 
       // If there were expensive iterations add them after a signalling -1.
       if ((solver_control_cheap.last_check() == dealii::SolverControl::failure)
-          && (solver_control_cheap.last_step() == solver_control_cheap.max_steps()))
+          && (solver_control_cheap.last_step() == solver_control_cheap.max_steps())
+          && (solver_control_expensive.max_steps() > 0))
```
Comment (bangerth) [inline source/postprocess/stokes_residual.cc]: Would we not get into the same kind of trouble is the cheap solver did not run?

**Comment by tjhei** [comment]
Rene, how does your change interact with the overriden ``get_history_data`` in compat.h? In other words, does this work with deal.II 8.5 and 9.0.pre?

**Comment by bangerth** [comment]
Ah, I misunderstood the code. Thanks for fixing this issue!

**Comment by gassmoeller** [comment]
@tjhei: I have not tested 9.0 extensively, but the `get_history_data()` functions are identical, and a quick run of the test showed no difference in the output, so I am reasonably confident it works with both versions.

**Comment by tjhei** [comment]
One more question: is this somehow related to this bugfix:
https://github.com/geodynamics/aspect/pull/1600/files#diff-e6b5753f114df2a6f2169165e60d554eR64

I don't remember why I did that. This is part of #1600 and not merged yet.

**Comment by gassmoeller** [comment]
Yes, with this PR you will no longer need the resize. We considered doing the resize when I fixed the history_data in deal.II, but decided against it, because a user might think `I do not know how many iterations I need, so I will simply set max_steps to 1e9`, and then that resize will make a mess :smile:.

**Pull request merged**